### PR TITLE
Make price oracle's external price sources available

### PR DESCRIPTION
**Pull request by e00E**
This is going to be used by the price estimator.

### Test Plan
No logic change.

### CODE REVIEW

File: core/src/price_estimation.rs [hunk 1]
```diff
@@ -160,6 +143,30 @@ impl PriceEstimating for PriceOracle {
...
+    token_info_fetcher: Arc<dyn TokenInfoFetching>,
+    update_interval: Duration,
+) -> Result<Vec<Box<dyn PriceSource + Send + Sync>>> {
+    let kraken = KrakenClient::new(http_factory, token_info_fetcher.clone())?;
+    let dexag = DexagClient::new(http_factory, token_info_fetcher.clone())?;
+    let oneinch = OneinchClient::new(http_factory, token_info_fetcher.clone())?;
+    Ok(vec![
+        thread_and_box(kraken, token_info_fetcher.clone(), update_interval),
+        thread_and_box(dexag, token_info_fetcher.clone(), update_interval),
+        thread_and_box(oneinch, token_info_fetcher.clone(), update_interval),
```
Comment (fleupold) [inline core/src/price_estimation.rs:158]: nanonit: last clone not necessary.

File: core/src/price_estimation.rs [hunk 2]
```diff
@@ -61,27 +61,10 @@ impl PriceOracle {
...
-            OneinchClient::new(http_factory, token_info_fetcher.clone())?,
-            update_interval,
-        );
-        let averaged_source = Box::new(AveragePriceSource::new(vec![
-            Box::new(kraken_source),
-            Box::new(dexag_source),
-            Box::new(oneinch_source),
-            Box::new(PricegraphEstimator::new(orderbook_reader)),
-        ]));
+        let mut external_price_sources =
```
Comment (fleupold) [inline core/src/price_estimation.rs:64]: ubernit: since we add an internal one, I'd call these just price_sources.

**Review by fleupold — APPROVED** [review]
How is the price estimator using these?

**Comment by e00E** [comment]
> How is the price estimator using these?

For the rounding buffer we need to know roughly the prices the solver uses. These prices sources are used by the solver so we also want them in price estimator.

**Review by nlordell — APPROVED** [review]

**Comment by fleupold** [comment]
> These prices sources are used by the solver so we also want them in price estimator.

But then we want to add the pricegraph one manually because of the way the threading works?

**Comment by e00E** [comment]
Yes

**Pull request merged**